src/api_midlevel.jl

```diff
@@ -46,6 +46,96 @@ function get_extent_dims(obj::Union{Dataspace,Dataset,Attribute})
     return dims, maxdims
 end
 
+"""
+    HDF5.get_extent_ndims(obj::Union{HDF5.Dataspace, HDF5.Dataset, HDF5.Attribute}) -> ndims
+
+Get the number of array dimensions from a dataspace, dataset, or attribute.
+
+"""
+function get_extent_ndims(obj::Union{Dataspace,Dataset,Attribute})
+    dspace = obj isa Dataspace ? checkvalid(obj) : dataspace(obj)
+    h5_ndims = h5s_get_simple_extent_ndims(dspace)
+    obj isa Dataspace || close(dspace)
+    return h5_ndims
+end
+
+"""
+    HDF5.get_chunk_offset(dataset_id, index)
+
+    Get 0-based offset of chunk from 0-based index.

```

Comment:
The docstring description below the method signature should not be indented (both here and below).

Reply:
I inundented the the docstring descriptions in 40e6279